public/framework/catalog_cache/cache_loader.tsx

```diff
@@ -164,12 +164,14 @@ export const updateTableColumnsToCache = (
     ts.name === tableName ? { ...ts, columns: tableColumns } : { ...ts }
   );
 
-  CatalogCacheManager.updateDatabase(dataSourceName, {
-    ...cachedDatabase,
-    tables: newTables,
-    lastUpdated: currentTime,
-    status: CachedDataSourceStatus.Updated,
-  });
+  if (cachedDatabase.status === CachedDataSourceStatus.Updated) {
```

Comment:
Can remove this line have a more covered checked on my PR: https://github.com/opensearch-project/dashboards-observability/pull/1617